### ISSUE TITLE
aws-custom-resource fails when `sdk-api-metadata.json` is missing

### ISSUE THREAD

**Issue by ahmetsevki** [aws/aws-cdk#7284]
## :question: General Issue
Can we make `sdk-api-metadata.json` parsing in `aws-custom-resource` lazy?

### The Question

I have a build process that compiles typescript code into js. And then I am packaging my lambda using `parcel.js`. I know I should be careful, however, cdk constructs leak into the generated code. The lambda fails at this location:
https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts#L11

This happens even before getting into my code. Somehow, I can't strip this module from the generated js. When modules are loaded it tries to find `sdk-api-metadata.json`, and fails so throws an exception. 

### Environment

  - **CDK CLI Version:** 
  - **Module Version:** 
  - **OS:** 
  - **Language:** 

### Other information

Possible solution:
```typescript

// don't use "require" since the typescript compiler emits errors since this
// file is not listed in tsconfig.json.
export type AwsSdkMetadata = { [key: string]: unknown };

const metadata = (): AwsSdkMetadata =>
  JSON.parse(fs.readFileSync(path.join(__dirname, 'sdk-api-metadata.json'), 'utf-8'));

// ....
function awsSdkToIamAction(service: string, action: string): string {
  const srv = service.toLowerCase();
 const awsSdkMetadata = metadata();
  const iamService = (awsSdkMetadata[srv] && awsSdkMetadata[srv].prefix) || srv;
  const iamAction = action.charAt(0).toUpperCase() + action.slice(1);
  return `${iamService}:${iamAction}`;
}


```

**Comment by eladb** [comment]
Isn't this an issue with your parcel configuration that causes it not to bring in this .json file?

I guess we can change this to be lazy, but I am wondering if that's the correct solution because this will then fail when actually trying to use `AwsCustomResource`.

@jogold what do you think?

We can also consider converting this .json file to a .js file, which will make it play better within the javascript module.

**Comment by jogold** [comment]
@ahmetsevki not sure I fully understand your setup. Why/how are your bundling the `AwsCustomResource` construct? What is your entry point and how does parcel get to that construct from there? What do you mean with _leak into the generated code_?

related if we change how metadata is included/patched: #4533, https://github.com/aws/aws-cdk/issues/6807#issuecomment-606512020

**Comment by ahmetsevki** [comment]
I have a framework, which allows you to write lambdas in typescript and then create a cdk stack that pushes the lambda to aws among other things.
 The framework code uses `aws-cdk` libraries and returns me an object with two functions: ` generateHandlers()`, and `generateStack()`. The first one `generateHandler` is used in subproject A and takes in a function and converts it to a lambda handler (`index.handler` and type checked events.) Then I build this project to a `dist` folder, and then I package it, using `parceljs` with such a command `parcel build dist/index.js --target=node --bundle-node-modules`.
The second one, `generateStack` is used in another  subproject that creates the cloudformation stack, and creates a lambda function using:
```typescript
import * as lambda from '@aws-cdk/aws-lambda';
const l = lambda.Function = new lambda.Function(app, 'fooFn', {   
  code: lambda.Code.fromAsset(`the folder where I packaged my lambda`),
```
The problem I am having is, since my framework generates code that does both the cloudformation stack creation and is also used in to create the lambda code, it contains `aws-cdk` libraries. When lambda code is starting up, it crashes with:
```
"errorMessage": "ENOENT: no such file or directory, open '/var/task/sdk-api-metadata.json'",
```
Because when lambda is starting, through cdk code `require`s that get into my `dist`, a `aws-custom-resource.ts` is `require`d (though unused). 
I looked into the code and I see that the 
`const metadata = JSON.parse(fs.readFileSync(path.join(__dirname, 'sdk-api-metadata.json'), 'utf-8'));` call in https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts#L11 is trying to read a `'sdk-api-metadata.json'` file that is not there. 
My suggestion was this could be made into a function rather than a const that is read at the module startup.
